### PR TITLE
Added /weather.

### PR DESCRIPTION
**Pull request by spacechase0**
Signed-off-by: Chase W chasew@silkweaver.com

### CODE REVIEW

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java [hunk 1]
```diff
@@ -173,39 +176,45 @@ public void xp(CommandContext args, CommandSource source) throws CommandExceptio
...
+			source.sendMessage("Weather set to RAIN.");
+			weather = Weather.RAIN;
+		} else if (args.getString(0).equalsIgnoreCase("thunderstorm")) {
+			source.sendMessage("Weather set to THUNDERSTORM.");
+			weather = Weather.THUNDERSTORM;
+		} else {
+			throw new CommandException("Weather must be a mode between 0 and 2, 'CLEAR', 'RAIN', or 'THUNDERSTORM'");
 		}
+		
+		Sky sky = plugin.getSky(plugin.spawnWorld);
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java]: Don't just use the spawn world, make sure the CommandSource is a player then use the world the player is in <code>player.getEntity().getWorld()</code>

**Comment by windy1** [comment]
Could you fix the thing I mentioned above? Looking good otherwise :)

**Comment by spacechase0** [comment]
Done.

I have no idea how to squash commits, but I'll have to do that tomorrow.

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java [hunk 2]
```diff
@@ -173,39 +176,51 @@ public void xp(CommandContext args, CommandSource source) throws CommandExceptio
...
-			} else if (args.getString(0).equalsIgnoreCase("snow")) {
-				source.sendMessage("Weather set to SNOW.");
-				// TODO: Switch weather to snow.
-			} else if (args.getString(0).equalsIgnoreCase("lightning")) {
-				source.sendMessage("Weather set to LIGHTNING.");
-				// TODO: Switch weather to lightning.
-			} else {
-				throw new CommandException("Weather must be a mode between 1 and 3, 'RAIN', 'SNOW', or 'LIGHTNING'");
+		if (!(source instanceof Player)) {
+			throw new CommandException("You must be a player to set the time!");
```
Comment (Zidane) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:184]: Console should be able to set the time of a world. Plus...I thought this was weather :S.

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java [hunk 3]
```diff
@@ -173,39 +176,51 @@ public void xp(CommandContext args, CommandSource source) throws CommandExceptio
...
+			source.sendMessage("Weather set to CLEAR.");
+			weather = Weather.CLEAR;
+		} else if (args.getString(0).equalsIgnoreCase("rain")) {
+			source.sendMessage("Weather set to RAIN.");
+			weather = Weather.RAIN;
+		} else if (args.getString(0).equalsIgnoreCase("thunderstorm")) {
+			source.sendMessage("Weather set to THUNDERSTORM.");
+			weather = Weather.THUNDERSTORM;
+		} else {
+			throw new CommandException("Weather must be a mode between 0 and 2, 'CLEAR', 'RAIN', or 'THUNDERSTORM'");
```
Comment (Zidane) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:217]: Y u no like Snow?

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java (hunk 2 above)
Comment (spacechase0) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:184]: It's like that for weather, too. I originally had it like that, @W1ndwaker told me to. :P

I must have forgotten to change it to weather when I copy+pasted it from the time command. :P

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java (hunk 3 above)
Comment (spacechase0) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:217]: I think snow and rain are the same thing, but snow is shown instead client-side for cold biomes.

Comment (Zidane) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:217]: I meant the text xD.

Comment (spacechase0) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:217]: Oh. I didn't think about that. :P

**Comment by spacechase0** [comment]
I'm going to make a new pull request. I messed that branch up. :P

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java (hunk 2 above)
Comment (windy1) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:184]: @NinjaZidane How would you know what world to set the time of? If we want this we need a world name param in the command.

Comment (spacechase0) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:184]: I'm adding it to the time command, too. :)

**Pull request closed without merging**